Sources/BraveWallet/WalletAssetBalanceManager.swift

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Data
+import BraveCore
+import Preferences
+import CoreData
+
+public protocol WalletUserAssetBalanceManagerType: AnyObject {
+  /// Return balance in String of the given asset. Return nil if there no balance stored
+  func getBalance(for asset: BraveWallet.BlockchainToken?, account: String?) -> [WalletUserAssetBalance]?
+  /// Store asset balance if there is none exists. Update asset balance if asset exists in database
+  func updateBalance(for asset: BraveWallet.BlockchainToken, account: String, balance: String, completion: (() -> Void)?)
+  /// Remove a `WalletUserAssetBalance` representation of the given
+  /// `BraveWallet.BlockchainToken` from CoreData
+  func removeBalance(for asset: BraveWallet.BlockchainToken, completion: (() -> Void)?)
+}
+
+public class WalletUserAssetBalanceManager: WalletUserAssetBalanceManagerType {
+ 
+  private let keyringService: BraveWalletKeyringService
+  private let rpcService: BraveWalletJsonRpcService
+  private let assetManager: WalletUserAssetManager
+  
+  init(
+    keyringService: BraveWalletKeyringService,
+    rpcService: BraveWalletJsonRpcService,
+    assetManager: WalletUserAssetManager
+  ) {
+    self.keyringService = keyringService
+    self.rpcService = rpcService
+    self.assetManager = assetManager
+  }
+  
+  public func getBalance(for asset: BraveWallet.BlockchainToken?, account: String?) -> [WalletUserAssetBalance]? {
+    WalletUserAssetBalance.getBalance(for: asset, account: account)
+  }
+  
+  public func updateBalance(for asset: BraveWallet.BlockchainToken, account: String, balance: String, completion: (() -> Void)?) {
+    WalletUserAssetBalance.updateBalance(
+      for: asset,
+      balance: balance,
+      account: account,
+      completion: completion
+    )
+  }
+  
+  public func removeBalance(for asset: BraveWallet.BlockchainToken, completion: (() -> Void)?) {
+    WalletUserAssetBalance.removeBalance(
+      for: asset,
+      completion: completion
+    )
+  }
+  
+  public func refreshBalances() async {
```

Comment:
I believe this should be called when `Preferences.Wallet.showTestNetworks` changes to true/enabled? I had to relaunch to see test network balance in Portfolio

Reply:
`WalletUserAssetManager` will observe `Preferences.Wallet.showTestNetworks` and boardcast its data observer when preference did changed.